controllers/machine_controller.go

```diff
@@ -598,6 +617,61 @@ func (r *MachineReconciler) shouldAdopt(m *clusterv1.Machine) bool {
 	return metav1.GetControllerOf(m) == nil && !util.HasOwner(m.OwnerReferences, clusterv1.GroupVersion.String(), []string{"Cluster"})
 }
 
+func (r *MachineReconciler) watchClusterNodes(ctx context.Context, cluster *clusterv1.Cluster) error {
+	// If there is no tracker, don't watch remote nodes
+	if r.Tracker == nil {
```

Comment:
Do we ever expect to support running without a tracker? Should this trow an error instead?

Reply:
It was for unit tests, which I think we still have some around